tools/nblIntrinicGenerator/example input/nabla.intrinsics.core.json

```diff
@@ -0,0 +1,55 @@
+{
+    "operand": [
+        {
+            "class": "Atomic",
+            "operandList": [
+                "*"
+            ]
+        },
+        {
+            "class": "Conversion",
+            "operandList": [
+                "OpBitcast"
+            ]
+        },
+        {
+            "class": "Bit",
+            "operandList": [
+                "OpBitFieldUExtract",
+                "OpBitFieldSExtract"
+            ]
+        }
+    ],
+    "builtins": [
+        {
+            "name": "HelperInvocation",
+            "type": "bool"
+        },
+        {
+            "name": "Position",
+            "type": "float32_t4",
+            "const": false
+        },
+        {
+            "name": "VertexIndex"
+        },
+        {
+            "name": "InstanceIndex"
+        },
+        {
+            "name": "NumWorkgroups"
+        },
+        {
+            "name": "WorkgroupId"
+        },
+        {
+            "name": "LocalInvocationId"
+        },
+        {
+            "name": "GlobalInvocationId"
+        },
+        {
+            "name": "LocalInvocationIndex"
+        }
+    ]
```

Comment:
we should export everything, and just add `type` and `const` fields to the grammar json in this repo

Reply:
ok I get the point of this... somewhat (it limits what we export)

but I think `type` and `const` should be added directly to the json enties in `spirv.core.grammar.json`, basically the `builtins` section shouldn't exist, unless you want to just list (allow/block) the builtin names you wish to export

Reply:
There are benefits to keeping type mapping in a separate file. 
The biggest one i can think of is that by modifying the grammar file, any subseqent updates pulled from Khronos Group might result in having to solve merge conflicts.

What might also happen is the c++ code generator stops working once the JSON schema changes, but surely its just me being paranoid.

Reply:
first one is moot

> What might also happen is the c++ code generator stops working once the JSON schema changes, but surely its just me being paranoid.

test this, but we'll only ever add extra fields